### PR TITLE
More refactoring3

### PR DESCRIPTION
**Pull request by seanmajorpayne**
Please note all changes:

- Removed old code from simple_scoring
- Temporary "actionHeadline" added for frontend
- Rewrote make_graph compute to account for all situations
- Recommended user Nodes now accounts for None by assigning -inf if None found.
- Refactored score_nodes.py for separation of concerns
- Moved json files into their own folder
- Added test_endpoints.py as a manual test for backend until Gabrielle can create more robust version

### CODE REVIEW

File: Dockerfile
Comment (rodriguesk) [inline Dockerfile:50]: This should not be changed here. It should be changed in the docker-compose.yml

File: api_tests/test_endpoints.py
Comment (rodriguesk) [inline api_tests/test_endpoints.py:6]: This url made need to to change to something like localhost... or 0.0.0.0... for it to work in circleCI. No need to change now but just a heads up for that potential

File: climatemind.py
Comment (rodriguesk) [inline climatemind.py:11]: Is there supposed to be a comma after Sessions?

File: docker-compose.yml
Comment (rodriguesk) [inline docker-compose.yml:14]: The “mysql:1433” part should not say MySQL, it should say something like "vc-db:1433" or “db:1433” I think. Not big deal but can be misleading otherwise. maybe “mssql-db:1433” would work, just need to try it before changing.

File: entrypoint.sh
Comment (rodriguesk) [inline entrypoint.sh:4]: This sleep should only happen when it’s localhost. No sleeps should be needed for production docker since the db is always on. 

This might be ok to merge now, but after merging if we notice the webapp makes user wait for 30 seconds a lot of times, this is why.

We only want sleeps happening in the docker-compose.yml. No sleeps should be needed to be triggered in the Dockerfile because they will then be triggered in the production container.

File: knowledge_graph/routes.py
Comment (rodriguesk) [inline knowledge_graph/routes.py:68]: For computer file paths they can’t be coded with “/“ because it won’t work on windows file systems (will break for devs doing localhost on windows).

Must replace everywhere in the code any file paths with “/“ (this doesn’t apply for urls though!).
Replace each one using the format:

    path_corrected = os.path.join(directory, child_folder, file_name) 

Don’t forget to also import os at the top of the file too

File: knowledge_graph/routes.py
Comment (rodriguesk) [inline knowledge_graph/routes.py:210]: Fix this path using os.path.join()

**Review by rodriguesk — CHANGES_REQUESTED** [review]
Please address all the single line comments made. 
Also please confirm all the files in the migrations folder are identical to those currently in the master branch (otherwise the Azure database will likely break since it’s currently working fine).

File: climatemind.py
Comment (seanmajorpayne) [reply climatemind.py:11]: commas after list are okay if they are on their own line. Very handy trick to remember. Makes it easier to add items/comment out other items in the future.

File: entrypoint.sh
Comment (seanmajorpayne) [reply entrypoint.sh:4]: Should only happen once on launch, but we can remember for the future if we have any issues.

**Comment by seanmajorpayne** [comment]
All changes have been fixed (except sleeps which temporarily will not move so we don't break the app)

File: Dockerfile
Comment (rodriguesk) [inline Dockerfile:50]: Roland's production code (his docker-compose.yml file for production) may expect we have EXPOSE 5000 8050 in the Dockerfile. If this merge breaks the production webapp it's likely because of this. Just keep this in mind since removing all the ports from being exposed in the Dockerfile.

File: api_tests/test_endpoints.py
Comment (rodriguesk) [inline api_tests/test_endpoints.py:7]: I'm pretty sure this still won't work on windows. Literally CANNOT use "/" in all paths without using special libraries. 
To get to previous directory have to use os functions to do so.  I believe what you want is `os.path.join(os.getcwd(), os.getcwd(os.pardir),"json_files"`

File: api_tests/test_endpoints.py
Comment (rodriguesk) [inline api_tests/test_endpoints.py:11]: I'm pretty sure the "/" should be removed here so it looks like `os.path.join(target_directory, "sample_user_response.json")`

File: knowledge_graph/routes.py
Comment (rodriguesk) [inline knowledge_graph/routes.py:69]: this needs fixing too for good style. 
Should be:
`file = os.path.join(os.getcwd(), "json_files","schwartz_questions.json")`

File: knowledge_graph/routes.py
Comment (rodriguesk) [inline knowledge_graph/routes.py:212]: this should also be:
`file = os.path.join(os.getcwd(), "json_files","value_descriptions.json")`

**Review by rodriguesk — APPROVED** [review]

**Review by rodriguesk — APPROVED** [review]

**Pull request merged**
